homeassistant/components/microbees/config_flow.py

```diff
@@ -0,0 +1,76 @@
+"""Config flow for microBees integration."""
+from collections.abc import Mapping
+import logging
+from typing import Any
+
+from microBeesPy.microbees import MicroBees, MicroBeesException
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_ACCESS_TOKEN, CONF_TOKEN
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import aiohttp_client, config_entry_oauth2_flow
+
+from .const import DOMAIN
+
+
+class OAuth2FlowHandler(
+    config_entry_oauth2_flow.AbstractOAuth2FlowHandler, domain=DOMAIN
+):
+    """Handle a config flow for microBees."""
+
+    DOMAIN = DOMAIN
+    reauth_entry: config_entries.ConfigEntry | None = None
+
+    @property
+    def logger(self) -> logging.Logger:
+        """Return logger."""
+        return logging.getLogger(__name__)
+
+    @property
+    def extra_authorize_data(self) -> dict[str, Any]:
+        """Extra data that needs to be appended to the authorize url."""
+        scopes = ["read", "write"]
+        return {"scope": " ".join(scopes)}
+
+    async def async_oauth_create_entry(self, data: dict[str, Any]) -> FlowResult:
+        """Create an oauth config entry or update existing entry for reauth."""
+
+        microbees = MicroBees(
+            session=aiohttp_client.async_get_clientsession(self.hass),
+            token=data[CONF_TOKEN][CONF_ACCESS_TOKEN],
+        )
+
+        try:
+            current_user = await microbees.getMyProfile()
+        except MicroBeesException:
+            return self.async_abort(reason="invalid_auth")
+        except Exception:  # pylint: disable=broad-except
+            return self.async_abort(reason="unknown")
```

Comment:
You still want to have a logger here, but you should use `self.logger` like said before.